src/Uno.UI/UI/Xaml/Window/Window.cs

```diff
@@ -0,0 +1,361 @@
+#pragma warning disable RS0030 // Current is banned
+	/// <summary>
+	/// Use this instead of Window.Current throughout this codebase
+	/// to prove it is intentional (the property is null throughout Uno.WinUI).
+	/// </summary>
+	internal static Window? CurrentSafe => _current;
+#pragma warning restore RS0030
+
+	/// <summary>
+	/// Use this only as a temporary measure on single-window targets.
+	/// </summary>
+	internal static Window? InitialWindow { get; private set; }
+
+	internal object? NativeWindow => _windowImplementation.NativeWindow;
+
+	/// <summary>
+	/// This is run when Application.Current is set and the UI framework is ready to construct
+	/// visual elements (this is important for eample for Andorid where trying to construct
+	/// UI during Application ctor will fail).
+	/// </summary>
+	internal void Initialize()
+	{
+		if (Application.Current?.InitializationComplete != true)
+		{
+			throw new InvalidOperationException("Application was not yet initialized.");
+		}
+
+		if (_initialized)
+		{
+			return;
+		}
+
+		_initialized = true;
+
+		if (_windowType is WindowType.CoreWindow)
+		{
+			WinUICoreServices.Instance.InitCoreWindowContentRoot();
+		}
+
+		_windowImplementation.Initialize();
+
+#if !HAS_UNO_WINUI
+		RaiseCreated();
+#endif
+	}
+
+	internal static void EnsureWindowCurrent()
+	{
+		if (_current is not null)
+		{
+			return;
+		}
+
+		_current = new Window(WindowType.CoreWindow);
+		InitialWindow = _current;
+	}
+
+	/// <summary>
+	/// Gets the CoreDispatcher object for the Window, which is generally the CoreDispatcher for the UI thread.
+	/// </summary>
+	public CoreDispatcher Dispatcher => CoreDispatcher.Main;
+
+#if HAS_UNO_WINUI
+	public global::Microsoft.UI.Dispatching.DispatcherQueue DispatcherQueue => global::Microsoft.UI.Dispatching.DispatcherQueue.Main;
+#else
+	internal global::Windows.System.DispatcherQueue DispatcherQueue => global::Windows.System.DispatcherQueue.Main;
+#endif
+
+	/// <summary>
+	/// Gets a value that reports whether the window is visible.
+	/// </summary>
+	public bool Visible => _windowImplementation.Visible;
+
+	/// <summary>
+	/// This is the real root of the **managed** visual tree.
+	/// This means its the root panel which contains the <see cref="Content"/>
+	/// but also the PopupRoot, the DragRoot and all other internal UI elements.
+	/// On platforms like iOS and Android, we might still have few native controls above this.
+	/// </summary>
+	/// <remarks>This element is flagged with IsVisualTreeRoot.</remarks>
+	internal UIElement? RootElement => _windowImplementation.XamlRoot?.VisualTree?.RootElement;
+
+	internal PopupRoot? PopupRoot => _windowImplementation.XamlRoot?.VisualTree?.PopupRoot;
+
+	internal FullWindowMediaRoot? FullWindowMediaRoot => _windowImplementation.XamlRoot?.VisualTree?.FullWindowMediaRoot;
+
+	internal Canvas? FocusVisualLayer => _windowImplementation.XamlRoot?.VisualTree?.FocusVisualRoot;
+
+	public void Activate()
+	{
+		_windowImplementation.Activate();
```

Comment:
This does not fail on android (no message or visible exception)

Reply:
Added `InvalidOperationException` in the `Window.cs` constructor on attempt to create secondary windows on unsupported targets.